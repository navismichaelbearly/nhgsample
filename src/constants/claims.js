export const permissions = 'http://nhhg.org.uk/claims/permissions';
export const userType = 'http://nhhg.org.uk/claims/user_type';
export const roles = 'http://nhhg.org.uk/claims/roles';
