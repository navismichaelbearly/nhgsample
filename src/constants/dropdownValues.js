export const TASKS_SEL = 'Tasks';
export const ARREARS_SEL = 'Arrears';
export const PRIORITY_SEL = 'priority';
export const CURRENT_BALANCE_SEL = 'currentBalance';
export const DAYS_IN_ARREARS_SEL = 'daysInArrears';
export const MY_TASKS_SEL = 'myTasks';
export const USER_TASKS_SEL = 'selectedUserTasks';
