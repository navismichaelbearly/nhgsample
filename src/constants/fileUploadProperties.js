const properties = Object.freeze({
  acceptedFileExtensions:
    '.doc,.docx,.xls,.xlsx,.ppt,.pptx,.pdf,.txt,.jpg,.png,.rtf,.htm,.html,.msg,.eml,.mht,.ics,.csv,.oft',
  maxFileSizeBytes: 1024 * 1024 * 20, // 20mb
  maxFilesDefault: 10,
  maxFilesNotes: 1,
});

export default properties;
