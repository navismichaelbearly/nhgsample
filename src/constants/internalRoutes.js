import { ACTIVE, DRAFT } from './pause';
import { isUserSeniorManager } from '../util/userRoles';
import legalReferralStatus, { status as legalReferralCaseStatus } from './legalReferral';

export const getActivityHistoryItemRoute = (arrearsId, itemId) =>
  `/arrears-details/${arrearsId || ':arrearsId'}/history/${itemId || ':itemId'}`;

export const getArrearsDetailRoute = arrearsId => `/arrears-details/${arrearsId || ':arrearsId'}`;

export const getArchivedTasksRoute = arrearsId =>
  `/arrears-details/${arrearsId || ':arrearsId'}/archived-tasks`;

export const getLegalReferralLink = ({
  arrearsId,
  legalReferralId,
  canCreateLegalReferral,
  status,
  userRoles,
}) => {
  const noLink = () => '';
  // pre-condition
  if (!arrearsId) {
    return noLink();
  }
  // helpers
  const prependedURL = `/arrears-details/${arrearsId}`;
  const reviewLink = () => `${prependedURL}/legal-case-referral/${legalReferralId}/review`;
  const approveLink = () => `${prependedURL}/legal-case-referral/${legalReferralId}/approve`;
  const createLink = () => `${prependedURL}/legal-case-referral/create`;
  const isLoggedUserSeniorManager = isUserSeniorManager(userRoles);
  // Legal Referral Task
  if (!legalReferralId) {
    if (canCreateLegalReferral && !isLoggedUserSeniorManager) {
      return createLink();
    }
    return noLink();
  }
  // Owner Sign-Off Task
  if (status === legalReferralCaseStatus.pendingSubmission) {
    if (isLoggedUserSeniorManager) {
      return approveLink();
    }
    return reviewLink();
  }
  // Manager Sign-Off
  if (status === legalReferralCaseStatus.requiresApproval) {
    return approveLink();
  }
  return reviewLink();
};

export const getTaskActivityRouteByActionType = ({
  actionType,
  arrearsId,
  legalReferral = {
    legalReferralId: '',
    canCreateLegalReferral: false,
    status: '',
    userRoles: [],
  },
}) => {
  const prependedURL = `/arrears-details/${arrearsId}`;
  switch (actionType) {
    case 'addinteraction': {
      return `${prependedURL}/interaction/create`;
    }
    case 'addnote': {
      return `${prependedURL}/note/create`;
    }
    case 'sendcorrespondence': {
      return `${prependedURL}/send-correspondence/create`;
    }
    case 'legalreferral': {
      return getLegalReferralLink({
        arrearsId,
        legalReferralId: legalReferral.legalReferralId,
        canCreateLegalReferral: legalReferral.canCreateLegalReferral,
        status: legalReferral.status,
        userRoles: legalReferral.userRoles,
      });
    }
    default: {
      return '#';
    }
  }
};

export default (
  { id: arrearsId, legalReferral: lr, legalReferralCase, pauseSummary, paymentPlan: pp },
  isManager
) => {
  // paymentPlan routing conditions
  const paymentPlanId = pp ? pp.id : null;

  // pause routing conditions
  const pauses = pauseSummary || [];
  const pauseAwaitingApproval = pauses.find(pause => pause.status === DRAFT);
  const activePause = pauses.find(pause => pause.status === ACTIVE);
  const currentPause = pauseAwaitingApproval || activePause || {};
  let pauseAction;
  if (currentPause.id) pauseAction = 'update';
  if (pauseAwaitingApproval && isManager) pauseAction = 'approve';

  // legalReferral routing conditions
  const legalReferralActionStatus = legalReferralCase ? legalReferralCase.status : null;
  const currentLegalReferralAction = lr ? lr.caseId : null;
  const legalReferralActionType =
    legalReferralActionStatus === legalReferralStatus.requiresApproval ? 'approve' : 'review';

  const linkToPage = (type, typeId, action) =>
    `/arrears-details/${arrearsId}/${type}/${typeId || 'create'}${
      typeId && action ? `/${action}` : ''
    }`;

  return {
    addNote: linkToPage('note'),
    addTask: linkToPage('task'),
    arrearsDetailNOSPModal: `/arrears-details/${arrearsId}?modal=serveNOSP`,
    interaction: linkToPage('interaction'),
    legalReferral: linkToPage(
      'legal-case-referral',
      currentLegalReferralAction,
      legalReferralActionType
    ),
    pause: linkToPage('pause-arrears-case', currentPause.id, pauseAction),
    paymentPlan: linkToPage('payment-plan', paymentPlanId),
    sendCorrespondence: linkToPage('send-correspondence'),
    tenancySustainmentReferral: linkToPage('refer-arrears-case'),
    welfareReformDetails: `/arrears-details/${arrearsId}/welfare-reform-details`,
  };
};
