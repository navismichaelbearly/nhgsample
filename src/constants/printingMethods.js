const LOCALPRINT = 'Local_Print';

export default LOCALPRINT;
