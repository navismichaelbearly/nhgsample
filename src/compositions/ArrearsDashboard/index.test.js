import React from 'react';
import { shallow } from 'enzyme';

import { ArrearsDashboardComposition } from './';
import TasksSummaryCards from '../../components/TasksSummaryCards';

import { TASKS_SEL, MY_TASKS_SEL, USER_TASKS_SEL } from '../../constants/dropdownValues';

describe('<ArrearsDashboardComposition />', () => {
  let props;
  let el;

  beforeEach(() => {
    props = {
      cardLabels: {
        address: 'Address',
        currentBalance: 'Current balance',
        heading: 'Arrears',
        notificationDate: 'Notification date',
        openTasks: 'Open tasks',
        tenantName: 'Tenant name',
      },
      taskCardLabels: {
        heading: 'Tasks',
        dueDate: 'Due Date',
        taskOwner: 'Task Owner',
        title: 'Title',
        customerName: 'Customer Name',
        address: 'Address',
        status: 'Status',
        overdue: 'overdue!',
      },
      arrears: [],
      tasks: [],
      myTasks: [],
      arrearsNoDataMsg: 'Gotham is in danger',
      myTasksNoDataMsg: 'My Gotham is in danger',
      tasksNoDataMsg: 'Gotham is in danger',
      heading: 'titleText',
      updatePageHeader: jest.fn(),
      getTasksSummary: jest.fn(),
      getArrearStatuses: jest.fn(),
      getTaskStatuses: jest.fn(),
      getSelectedUserTasks: jest.fn(),
      getSummaryTasks: jest.fn(),
      arrearStatuses: {},
      taskStatuses: {},
      loadingArrears: false,
      loadingTasks: false,
      profile: {
        id: 'ABC123',
        patchName: 'Hello world',
      },
      patches: [
        {
          name: 'John Smith',
          patchName: 'Hello world',
        },
        {
          name: 'Jane Smith',
          patchName: 'Some other name',
        },
      ],
      getMyTasksSummary: jest.fn(),
      getArrearsSummary: jest.fn(),
      onOpenPatchSelect: jest.fn(),
      invalidateTasks: jest.fn(),
      getUserTasksSummary: jest.fn(),
      getStatuses: jest.fn(),
      selectedUserTasks: [],
      selectedUserTasksNoDataMsg: '',
      viewMoreLoading: false,
    };

    el = shallow(<ArrearsDashboardComposition {...props} />);
  });

  it('should render the page', () => {
    expect(el).toMatchSnapshot();
  });

  it('selected user tasks with more result count', () => {
    el.setState({ statusFilterId: USER_TASKS_SEL });
    el.setProps({
      selectedUserTasksCount: 16,
      selectedUserTasks: {
        count: 16,
        items: [
          {
            id: 'd710fe3a-7142-481a-be10-0c0ba5e55dfa',
            type: 'manual',
            status: 'Open',
            title: 'Task 10 title',
            dueDate: '2021-01-21T18:30:00+00:00',
            assetName: 'abc',
            tenantName: 'xyz',
          },
          {
            id: '69b8964e-53ad-4d07-9e25-160caf52f3e6',
            type: 'manual',
            status: 'Open',
            title: 'Task 8',
            dueDate: '2021-01-21T18:30:00+00:00',
            assetName: 'abc',
            tenantName: 'xyz',
          },
          {
            id: '8ec08b38-cd64-4d3f-93bd-254afb8dd71e',
            type: 'manual',
            status: 'Open',
            title: 'Task 2',
            dueDate: '2021-01-21T18:30:00+00:00',
            assetName: 'abc',
            tenantName: 'xyz',
          },
          {
            id: 'ec433738-235f-40b5-8472-2623dafdc7b3',
            type: 'manual',
            status: 'Open',
            title: 'Task 4',
            dueDate: '2021-01-21T18:30:00+00:00',
            assetName: 'abc',
            tenantName: 'xyz',
          },
          {
            id: 'c18968ff-5ed6-4bc1-bd49-4ec9785c2767',
            type: 'manual',
            status: 'Open',
            title: 'Task 5',
            dueDate: '2021-01-28T18:30:00+00:00',
            assetName: 'abc',
            tenantName: 'xyz',
          },
          {
            id: 'f2f5b601-06a6-4439-ac64-797a058ef063',
            type: 'manual',
            status: 'Open',
            title: 'Tap Repair',
            dueDate: '2021-02-24T18:30:00+00:00',
            assetName: 'abc',
            tenantName: 'xyz',
          },
        ],
      },
    }).update();
    expect(el).toMatchSnapshot();
  });

  it('my tasks user tasks with more result count', () => {
    el.setState({ statusFilterId: MY_TASKS_SEL });
    el.setProps({
      myTasks: [
        {
          id: 'd710fe3a-7142-481a-be10-0c0ba5e55dfa',
          type: 'manual',
          status: 'Open',
          title: 'Task 10 title',
          dueDate: '2021-01-21T18:30:00+00:00',
          assetName: 'abc',
          tenantName: 'xyz',
        },
        {
          id: '69b8964e-53ad-4d07-9e25-160caf52f3e6',
          type: 'manual',
          status: 'Open',
          title: 'Task 8',
          dueDate: '2021-01-21T18:30:00+00:00',
          assetName: 'abc',
          tenantName: 'xyz',
        },
        {
          id: '8ec08b38-cd64-4d3f-93bd-254afb8dd71e',
          type: 'manual',
          status: 'Open',
          title: 'Task 2',
          dueDate: '2021-01-21T18:30:00+00:00',
          assetName: 'abc',
          tenantName: 'xyz',
        },
        {
          id: 'ec433738-235f-40b5-8472-2623dafdc7b3',
          type: 'manual',
          status: 'Open',
          title: 'Task 4',
          dueDate: '2021-01-21T18:30:00+00:00',
          assetName: 'abc',
          tenantName: 'xyz',
        },
        {
          id: 'c18968ff-5ed6-4bc1-bd49-4ec9785c2767',
          type: 'manual',
          status: 'Open',
          title: 'Task 5',
          dueDate: '2021-01-28T18:30:00+00:00',
          assetName: 'abc',
          tenantName: 'xyz',
        },
        {
          id: 'f2f5b601-06a6-4439-ac64-797a058ef063',
          type: 'manual',
          status: 'Open',
          title: 'Tap Repair',
          dueDate: '2021-02-24T18:30:00+00:00',
          assetName: 'abc',
          tenantName: 'xyz',
        },
      ],
    }).update();
    expect(el).toMatchSnapshot();
  });

  it('tasks patches with more result count', () => {
    el.setState({ statusFilterId: TASKS_SEL });
    el.setProps({
      tasks: [
        {
          id: 'd710fe3a-7142-481a-be10-0c0ba5e55dfa',
          type: 'manual',
          status: 'Open',
          title: 'Task 10 title',
          dueDate: '2021-01-21T18:30:00+00:00',
          assetName: 'abc',
          tenantName: 'xyz',
        },
        {
          id: '69b8964e-53ad-4d07-9e25-160caf52f3e6',
          type: 'manual',
          status: 'Open',
          title: 'Task 8',
          dueDate: '2021-01-21T18:30:00+00:00',
          assetName: 'abc',
          tenantName: 'xyz',
        },
        {
          id: '8ec08b38-cd64-4d3f-93bd-254afb8dd71e',
          type: 'manual',
          status: 'Open',
          title: 'Task 2',
          dueDate: '2021-01-21T18:30:00+00:00',
          assetName: 'abc',
          tenantName: 'xyz',
        },
        {
          id: 'ec433738-235f-40b5-8472-2623dafdc7b3',
          type: 'manual',
          status: 'Open',
          title: 'Task 4',
          dueDate: '2021-01-21T18:30:00+00:00',
          assetName: 'abc',
          tenantName: 'xyz',
        },
        {
          id: 'c18968ff-5ed6-4bc1-bd49-4ec9785c2767',
          type: 'manual',
          status: 'Open',
          title: 'Task 5',
          dueDate: '2021-01-28T18:30:00+00:00',
          assetName: 'abc',
          tenantName: 'xyz',
        },
        {
          id: 'f2f5b601-06a6-4439-ac64-797a058ef063',
          type: 'manual',
          status: 'Open',
          title: 'Tap Repair',
          dueDate: '2021-02-24T18:30:00+00:00',
          assetName: 'abc',
          tenantName: 'xyz',
        },
      ],
    }).update();
    expect(el).toMatchSnapshot();
  });

  it('arrears list validation', () => {
    el.setState({ statusFilterId: null });
    el.setProps({
      filterLabel: 'Arrears - System Paused (50)',
      resultCount: 6,
      arrears: [
        {
          createdOn: '2021-02-15T16:13:48+00:00',
          currentBalance: 0,
          daysInArrears: 16,
          id: 'abc1',
          lastModified: '2021-03-03T07:23:43+00:00',
          openTaskCount: 1,
          ownerOutputModel: {
            id: 'xyz1',
            name: 'Eileen O Neill',
            type: 'systemuser',
          },
          priority: 54361,
          status: 'System Paused',
          tenancy: {
            address: 'Flat 3 313 Harrow Road, London, 3RJ',
            id: 1,
            name: 'Dummy Pommells',
          },
          title: 'Arrears - 50101743',
        },
        {
          createdOn: '2021-02-15T16:13:48+00:00',
          currentBalance: 0,
          daysInArrears: 16,
          id: 'abc2',
          lastModified: '2021-03-03T07:23:43+00:00',
          openTaskCount: 1,
          ownerOutputModel: {
            id: 'xyz2',
            name: 'Eileen O Neill',
            type: 'systemuser',
          },
          priority: 54361,
          status: 'System Paused',
          tenancy: {
            address: 'Flat 3 313 Harrow Road, London, 3RJ',
            id: 2,
            name: 'Dummy Pommells',
          },
          title: 'Arrears - 50101743',
        },
        {
          createdOn: '2021-02-15T16:13:48+00:00',
          currentBalance: 0,
          daysInArrears: 16,
          id: 'abc3',
          lastModified: '2021-03-03T07:23:43+00:00',
          openTaskCount: 1,
          ownerOutputModel: {
            id: 'xyz3',
            name: 'Eileen O Neill',
            type: 'systemuser',
          },
          priority: 54361,
          status: 'System Paused',
          tenancy: {
            address: 'Flat 3 313 Harrow Road, London, 3RJ',
            id: 3,
            name: 'Dummy Pommells',
          },
          title: 'Arrears - 50101743',
        },
        {
          createdOn: '2021-02-15T16:13:48+00:00',
          currentBalance: 0,
          daysInArrears: 16,
          id: 'abc4',
          lastModified: '2021-03-03T07:23:43+00:00',
          openTaskCount: 1,
          ownerOutputModel: {
            id: 'xyz4',
            name: 'Eileen O Neill',
            type: 'systemuser',
          },
          priority: 54361,
          status: 'System Paused',
          tenancy: {
            address: 'Flat 3 313 Harrow Road, London, 3RJ',
            id: 4,
            name: 'Dummy Pommells',
          },
          title: 'Arrears - 50101743',
        },
        {
          createdOn: '2021-02-15T16:13:48+00:00',
          currentBalance: 0,
          daysInArrears: 16,
          id: 'abc5',
          lastModified: '2021-03-03T07:23:43+00:00',
          openTaskCount: 1,
          ownerOutputModel: {
            id: 'xyz5',
            name: 'Eileen O Neill',
            type: 'systemuser',
          },
          priority: 54361,
          status: 'System Paused',
          tenancy: {
            address: 'Flat 3 313 Harrow Road, London, 3RJ',
            id: 5,
            name: 'Dummy Pommells',
          },
          title: 'Arrears - 50101743',
        },
        {
          createdOn: '2021-02-15T16:13:48+00:00',
          currentBalance: 0,
          daysInArrears: 16,
          id: 'abc6',
          lastModified: '2021-03-03T07:23:43+00:00',
          openTaskCount: 1,
          ownerOutputModel: {
            id: 'xyz6',
            name: 'Eileen O Neill',
            type: 'systemuser',
          },
          priority: 54361,
          status: 'System Paused',
          tenancy: {
            address: 'Flat 3 313 Harrow Road, London, 3RJ',
            id: 6,
            name: 'Dummy Pommells',
          },
          title: 'Arrears - 50101743',
        },
      ],
    }).update();
    expect(el).toMatchSnapshot();
  });

  it('should render the page while loading arrears and tasks', () => {
    el.setProps({
      loadingArrears: true,
      loadingTasks: true,
    }).update();
    expect(el).toMatchSnapshot();
  });

  it('should update the header', () => {
    expect(props.updatePageHeader).toHaveBeenCalled();
  });

  it('should call getTaskSummary', () => {
    expect(props.getTasksSummary).toHaveBeenCalled();
  });

  it('should call invalidateTasks on UnMount', () => {
    el.unmount();
    expect(props.invalidateTasks).toHaveBeenCalled();
  });

  describe('Tasks - no data message', () => {
    beforeEach(() => {
      el.setProps({
        arrears: [
          {
            businessUnit: 'PRH',
            id: 'foo',
            bar: 'baz',
            legalReferralCase: {
              formName: 'Friendly Form Name | legalReferralCaseFormName',
              lastSubmissionId: 'DEF456',
            },
            nospServeError: false,
            tenancyId: 'ABC123',
            createdOn: '1',
          },
        ],
      });
    });

    it('should render tasksNoDataMsg patch tasks is selected and there are no tasks', () => {
      el.setState({ statusFilterId: TASKS_SEL });
      const cards = el.find(TasksSummaryCards);
      expect(cards).toExist();
      expect(cards.props()).toHaveProperty('noDataMsg', props.tasksNoDataMsg);
    });

    it('should render myTasksNoDataMsg when my tasks is selected and there are no tasks', () => {
      el.setState({ statusFilterId: MY_TASKS_SEL });
      const cards = el.find(TasksSummaryCards);
      expect(cards).toExist();
      expect(cards.props()).toHaveProperty('noDataMsg', props.myTasksNoDataMsg);
    });

    it('should render myTasksNoDataMsg when user tasks is selected and there are no tasks', () => {
      el.setState({ statusFilterId: USER_TASKS_SEL });
      const cards = el.find(TasksSummaryCards);
      expect(cards).toExist();
      expect(cards.props()).toHaveProperty('noDataMsg', props.selectedUserTasksNoDataMsg);
    });
  });
});
