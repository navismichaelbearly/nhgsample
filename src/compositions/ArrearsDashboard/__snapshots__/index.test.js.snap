// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrearsDashboardComposition /> arrears list validation 1`] = `
<withRouter(Connect(PageContentComposition))>
  <div
    className="col-lg-9"
  >
    <Connect(ArrearsFiltersComposition)
      onChange={[Function]}
    />
  </div>
  <div
    className="col-lg-9"
  >
    <WithTheme(Component)
      arrears={
        Array [
          Object {
            "createdOn": "2021-02-15T16:13:48+00:00",
            "currentBalance": 0,
            "daysInArrears": 16,
            "id": "abc1",
            "lastModified": "2021-03-03T07:23:43+00:00",
            "openTaskCount": 1,
            "ownerOutputModel": Object {
              "id": "xyz1",
              "name": "Eileen O Neill",
              "type": "systemuser",
            },
            "priority": 54361,
            "status": "System Paused",
            "tenancy": Object {
              "address": "Flat 3 313 Harrow Road, London, 3RJ",
              "id": 1,
              "name": "Dummy Pommells",
            },
            "title": "Arrears - 50101743",
          },
          Object {
            "createdOn": "2021-02-15T16:13:48+00:00",
            "currentBalance": 0,
            "daysInArrears": 16,
            "id": "abc2",
            "lastModified": "2021-03-03T07:23:43+00:00",
            "openTaskCount": 1,
            "ownerOutputModel": Object {
              "id": "xyz2",
              "name": "Eileen O Neill",
              "type": "systemuser",
            },
            "priority": 54361,
            "status": "System Paused",
            "tenancy": Object {
              "address": "Flat 3 313 Harrow Road, London, 3RJ",
              "id": 2,
              "name": "Dummy Pommells",
            },
            "title": "Arrears - 50101743",
          },
          Object {
            "createdOn": "2021-02-15T16:13:48+00:00",
            "currentBalance": 0,
            "daysInArrears": 16,
            "id": "abc3",
            "lastModified": "2021-03-03T07:23:43+00:00",
            "openTaskCount": 1,
            "ownerOutputModel": Object {
              "id": "xyz3",
              "name": "Eileen O Neill",
              "type": "systemuser",
            },
            "priority": 54361,
            "status": "System Paused",
            "tenancy": Object {
              "address": "Flat 3 313 Harrow Road, London, 3RJ",
              "id": 3,
              "name": "Dummy Pommells",
            },
            "title": "Arrears - 50101743",
          },
          Object {
            "createdOn": "2021-02-15T16:13:48+00:00",
            "currentBalance": 0,
            "daysInArrears": 16,
            "id": "abc4",
            "lastModified": "2021-03-03T07:23:43+00:00",
            "openTaskCount": 1,
            "ownerOutputModel": Object {
              "id": "xyz4",
              "name": "Eileen O Neill",
              "type": "systemuser",
            },
            "priority": 54361,
            "status": "System Paused",
            "tenancy": Object {
              "address": "Flat 3 313 Harrow Road, London, 3RJ",
              "id": 4,
              "name": "Dummy Pommells",
            },
            "title": "Arrears - 50101743",
          },
          Object {
            "createdOn": "2021-02-15T16:13:48+00:00",
            "currentBalance": 0,
            "daysInArrears": 16,
            "id": "abc5",
            "lastModified": "2021-03-03T07:23:43+00:00",
            "openTaskCount": 1,
            "ownerOutputModel": Object {
              "id": "xyz5",
              "name": "Eileen O Neill",
              "type": "systemuser",
            },
            "priority": 54361,
            "status": "System Paused",
            "tenancy": Object {
              "address": "Flat 3 313 Harrow Road, London, 3RJ",
              "id": 5,
              "name": "Dummy Pommells",
            },
            "title": "Arrears - 50101743",
          },
          Object {
            "createdOn": "2021-02-15T16:13:48+00:00",
            "currentBalance": 0,
            "daysInArrears": 16,
            "id": "abc6",
            "lastModified": "2021-03-03T07:23:43+00:00",
            "openTaskCount": 1,
            "ownerOutputModel": Object {
              "id": "xyz6",
              "name": "Eileen O Neill",
              "type": "systemuser",
            },
            "priority": 54361,
            "status": "System Paused",
            "tenancy": Object {
              "address": "Flat 3 313 Harrow Road, London, 3RJ",
              "id": 6,
              "name": "Dummy Pommells",
            },
            "title": "Arrears - 50101743",
          },
        ]
      }
      cardLabels={
        Object {
          "address": "Address",
          "currentBalance": "Current balance",
          "heading": "Arrears",
          "notificationDate": "Notification date",
          "openTasks": "Open tasks",
          "tenantName": "Tenant name",
        }
      }
      filterLabel="Arrears - System Paused (50)"
      getArrearsSummary={[Function]}
      loading={false}
      noDataMsg="Gotham is in danger"
      resultsCount={0}
      userId="ABC123"
      youText=""
    />
  </div>
  <aside
    className="col-lg-3"
  >
    <ErrorBoundary
      messageOnScreen={null}
    >
      <styled.div>
        <C
          disable={false}
          isButtonBottomAligned={true}
          openPatchSelect={[Function]}
          patches={
            Array [
              Object {
                "customName": false,
                "fullname": "John Smith",
                "name": "John Smith",
                "patchName": "Hello world",
              },
              Object {
                "customName": false,
                "fullname": "Jane Smith",
                "name": "Jane Smith",
                "patchName": "Some other name",
              },
            ]
          }
        />
      </styled.div>
    </ErrorBoundary>
    <Connect(PatchStatisticsCompositions) />
  </aside>
</withRouter(Connect(PageContentComposition))>
`;

exports[`<ArrearsDashboardComposition /> my tasks user tasks with more result count 1`] = `
<withRouter(Connect(PageContentComposition))>
  <div
    className="col-lg-9"
  >
    <Connect(ArrearsFiltersComposition)
      onChange={[Function]}
    />
  </div>
  <div
    className="col-lg-9"
  >
    <WithTheme(Component)
      cardLabels={
        Object {
          "address": "Address",
          "customerName": "Customer Name",
          "dueDate": "Due Date",
          "heading": "Tasks",
          "overdue": "overdue!",
          "status": "Status",
          "taskOwner": "Task Owner",
          "title": "Title",
        }
      }
      getNextPage={[Function]}
      key="myTasksSel"
      kind="myTasks"
      loading={false}
      noDataMsg="My Gotham is in danger"
      resultCount={0}
      tasks={Array []}
      userId="ABC123"
      youText=""
    />
  </div>
  <aside
    className="col-lg-3"
  >
    <ErrorBoundary
      messageOnScreen={null}
    >
      <styled.div>
        <C
          disable={false}
          isButtonBottomAligned={true}
          openPatchSelect={[Function]}
          patches={
            Array [
              Object {
                "customName": false,
                "fullname": "John Smith",
                "name": "John Smith",
                "patchName": "Hello world",
              },
              Object {
                "customName": false,
                "fullname": "Jane Smith",
                "name": "Jane Smith",
                "patchName": "Some other name",
              },
            ]
          }
        />
      </styled.div>
    </ErrorBoundary>
    <Connect(PatchStatisticsCompositions) />
  </aside>
</withRouter(Connect(PageContentComposition))>
`;

exports[`<ArrearsDashboardComposition /> selected user tasks with more result count 1`] = `
<withRouter(Connect(PageContentComposition))>
  <div
    className="col-lg-9"
  >
    <Connect(ArrearsFiltersComposition)
      onChange={[Function]}
    />
  </div>
  <div
    className="col-lg-9"
  >
    <WithTheme(Component)
      cardLabels={
        Object {
          "address": "Address",
          "customerName": "Customer Name",
          "dueDate": "Due Date",
          "heading": "Tasks",
          "overdue": "overdue!",
          "status": "Status",
          "taskOwner": "Task Owner",
          "title": "Title",
        }
      }
      getNextPage={[Function]}
      key="userTasksSel"
      kind="selectedUserTasks"
      loading={false}
      noDataMsg=""
      resultCount={16}
      tasks={
        Array [
          Object {
            "assetName": "abc",
            "dueDate": "2021-01-21T18:30:00+00:00",
            "id": "d710fe3a-7142-481a-be10-0c0ba5e55dfa",
            "status": "Open",
            "tenantName": "xyz",
            "title": "Task 10 title",
            "type": "manual",
          },
          Object {
            "assetName": "abc",
            "dueDate": "2021-01-21T18:30:00+00:00",
            "id": "69b8964e-53ad-4d07-9e25-160caf52f3e6",
            "status": "Open",
            "tenantName": "xyz",
            "title": "Task 8",
            "type": "manual",
          },
          Object {
            "assetName": "abc",
            "dueDate": "2021-01-21T18:30:00+00:00",
            "id": "8ec08b38-cd64-4d3f-93bd-254afb8dd71e",
            "status": "Open",
            "tenantName": "xyz",
            "title": "Task 2",
            "type": "manual",
          },
          Object {
            "assetName": "abc",
            "dueDate": "2021-01-21T18:30:00+00:00",
            "id": "ec433738-235f-40b5-8472-2623dafdc7b3",
            "status": "Open",
            "tenantName": "xyz",
            "title": "Task 4",
            "type": "manual",
          },
          Object {
            "assetName": "abc",
            "dueDate": "2021-01-28T18:30:00+00:00",
            "id": "c18968ff-5ed6-4bc1-bd49-4ec9785c2767",
            "status": "Open",
            "tenantName": "xyz",
            "title": "Task 5",
            "type": "manual",
          },
          Object {
            "assetName": "abc",
            "dueDate": "2021-02-24T18:30:00+00:00",
            "id": "f2f5b601-06a6-4439-ac64-797a058ef063",
            "status": "Open",
            "tenantName": "xyz",
            "title": "Tap Repair",
            "type": "manual",
          },
        ]
      }
      userId="ABC123"
      youText=""
    />
  </div>
  <aside
    className="col-lg-3"
  >
    <ErrorBoundary
      messageOnScreen={null}
    >
      <styled.div>
        <C
          disable={false}
          isButtonBottomAligned={true}
          openPatchSelect={[Function]}
          patches={
            Array [
              Object {
                "customName": false,
                "fullname": "John Smith",
                "name": "John Smith",
                "patchName": "Hello world",
              },
              Object {
                "customName": false,
                "fullname": "Jane Smith",
                "name": "Jane Smith",
                "patchName": "Some other name",
              },
            ]
          }
        />
      </styled.div>
    </ErrorBoundary>
    <Connect(PatchStatisticsCompositions) />
  </aside>
</withRouter(Connect(PageContentComposition))>
`;

exports[`<ArrearsDashboardComposition /> should render the page 1`] = `
<withRouter(Connect(PageContentComposition))>
  <div
    className="col-lg-9"
  >
    <Connect(ArrearsFiltersComposition)
      onChange={[Function]}
    />
  </div>
  <div
    className="col-lg-9"
  >
    <WithTheme(Component)
      cardLabels={
        Object {
          "address": "Address",
          "customerName": "Customer Name",
          "dueDate": "Due Date",
          "heading": "Tasks",
          "overdue": "overdue!",
          "status": "Status",
          "taskOwner": "Task Owner",
          "title": "Title",
        }
      }
      getNextPage={[Function]}
      key="myTasksSel"
      kind="myTasks"
      loading={false}
      noDataMsg="My Gotham is in danger"
      resultCount={0}
      tasks={Array []}
      userId="ABC123"
      youText=""
    />
  </div>
  <aside
    className="col-lg-3"
  >
    <ErrorBoundary
      messageOnScreen={null}
    >
      <styled.div>
        <C
          disable={false}
          isButtonBottomAligned={true}
          openPatchSelect={[Function]}
          patches={
            Array [
              Object {
                "customName": false,
                "fullname": "John Smith",
                "name": "John Smith",
                "patchName": "Hello world",
              },
              Object {
                "customName": false,
                "fullname": "Jane Smith",
                "name": "Jane Smith",
                "patchName": "Some other name",
              },
            ]
          }
        />
      </styled.div>
    </ErrorBoundary>
    <Connect(PatchStatisticsCompositions) />
  </aside>
</withRouter(Connect(PageContentComposition))>
`;

exports[`<ArrearsDashboardComposition /> should render the page while loading arrears and tasks 1`] = `
<withRouter(Connect(PageContentComposition))>
  <div
    className="col-lg-9"
  >
    <Connect(ArrearsFiltersComposition)
      onChange={[Function]}
    />
  </div>
  <div
    className="col-lg-9"
  >
    <p
      isSmall={false}
    />
  </div>
  <aside
    className="col-lg-3"
  >
    <ErrorBoundary
      messageOnScreen={null}
    >
      <styled.div>
        <C
          disable={false}
          isButtonBottomAligned={true}
          openPatchSelect={[Function]}
          patches={
            Array [
              Object {
                "customName": false,
                "fullname": "John Smith",
                "name": "John Smith",
                "patchName": "Hello world",
              },
              Object {
                "customName": false,
                "fullname": "Jane Smith",
                "name": "Jane Smith",
                "patchName": "Some other name",
              },
            ]
          }
        />
      </styled.div>
    </ErrorBoundary>
    <Connect(PatchStatisticsCompositions) />
  </aside>
</withRouter(Connect(PageContentComposition))>
`;

exports[`<ArrearsDashboardComposition /> tasks patches with more result count 1`] = `
<withRouter(Connect(PageContentComposition))>
  <div
    className="col-lg-9"
  >
    <Connect(ArrearsFiltersComposition)
      onChange={[Function]}
    />
  </div>
  <div
    className="col-lg-9"
  >
    <WithTheme(Component)
      cardLabels={
        Object {
          "address": "Address",
          "customerName": "Customer Name",
          "dueDate": "Due Date",
          "heading": "Tasks",
          "overdue": "overdue!",
          "status": "Status",
          "taskOwner": "Task Owner",
          "title": "Title",
        }
      }
      getNextPage={[Function]}
      key="tasksSel"
      kind="Tasks"
      loading={false}
      noDataMsg="Gotham is in danger"
      resultCount={0}
      tasks={Array []}
      userId="ABC123"
      youText=""
    />
  </div>
  <aside
    className="col-lg-3"
  >
    <ErrorBoundary
      messageOnScreen={null}
    >
      <styled.div>
        <C
          disable={false}
          isButtonBottomAligned={true}
          openPatchSelect={[Function]}
          patches={
            Array [
              Object {
                "customName": false,
                "fullname": "John Smith",
                "name": "John Smith",
                "patchName": "Hello world",
              },
              Object {
                "customName": false,
                "fullname": "Jane Smith",
                "name": "Jane Smith",
                "patchName": "Some other name",
              },
            ]
          }
        />
      </styled.div>
    </ErrorBoundary>
    <Connect(PatchStatisticsCompositions) />
  </aside>
</withRouter(Connect(PageContentComposition))>
`;
